addons/reorder-custom-inputs/userscript.js

```diff
@@ -78,21 +122,78 @@ export default async function ({ addon, console }) {
     }
   }
 
-  const originalShowEditor = Blockly.FieldTextInputRemovable.prototype.showEditor_;
+  function shiftFieldCallback(sourceBlock, field, direction) {
+    const proc = sourceBlock.parentBlock_ ? sourceBlock.parentBlock_ : sourceBlock;
+
+    // if inputList length is 1 there's nowhere to shift the input so we can simply return
+    if (proc.inputList.length <= 1) return;
+
+    const { name, index } = getFieldInputNameAndIndex(field, proc.inputList);
+    const newPosition = direction === "left" ? index - 1 : index + 1;
+    shiftInput(proc, name, newPosition);
+  }
+
+  function polluteProcedureDeclaration(procedureDeclaration, save_original = true) {
+    procedureDeclaration.createAllInputs_ = modifiedCreateAllInputs;
+    procedureDeclaration.onChangeFn = modifiedUpdateDeclarationProcCode;
+    procedureDeclaration.removeFieldCallback = modifiedRemoveFieldCallback;
+
+    for (const inputFn of ["addLabelExternal", "addBooleanExternal", "addStringNumberExternal"]) {
+      if (save_original) {
+        originalAddFns[inputFn] = procedureDeclaration[inputFn];
+      }
+      procedureDeclaration[inputFn] = addInputAfter(procedureDeclaration[inputFn], inputFn);
+    }
+  }
+
+  function depolluteProcedureDeclaration(procedureDeclaration) {
+    procedureDeclaration.createAllInputs_ = originalCreateAllInputs;
+    procedureDeclaration.onChangeFn = originalUpdateDeclarationProcCode;
+    procedureDeclaration.removeFieldCallback = originalRemoveFieldCallback;
+
+    for (const [inputFnName, originalFn] of Object.entries(originalAddFns)) {
+      procedureDeclaration[inputFnName] = originalFn;
+    }
+  }
 
   function enableAddon() {
+    // pollute the procedures_declaration prototype with a modified version that prevents merging, and allows inserting after
+    polluteProcedureDeclaration(Blockly.Blocks["procedures_declaration"]);
+
+    // if custom procedures modal is already open we also directly pollute the existing procedures_declaration block
+    if (addon.tab.redux.state.scratchGui.customProcedures.active) {
+      polluteProcedureDeclaration(Blockly.getMainWorkspace().getAllBlocks()[0], false);
```

Comment:
I've read a message by GarboMuffin on Discord where he mentions that `Blockly.getMainWorkspace()` is used intentionally here, and that it should not be changed to `addon.tab.traps.getWorkspace()` in the future. Could you add this as a code comment?

Reply:
For some context, the behavior of getWorkspace is usually preferred (that's why it exists). Some addons weren't updated yet to use it, and updating this usage of getMainWorkspace() would cause bugs, so I want to avoid future issues.

Reply:
I have added a comment